### PR TITLE
inital gulps integration

### PR DESCRIPTION
**Pull request by fireice-uk**
Doesn't build (fmt integration), but shows where I want to take the logging system.

I will probably split output levels from errors so we can define "debug level 2 error" and "output level 0 warning".

- [ ] WIP - do not merge.

### CODE REVIEW

**Comment by psychocrypt** [comment]
@fireice-uk  I added a commit with a literal operator to convert C strings into std::string and in an additional commit I fixed all compile issues.
By accident I pushed the commits from #77 into this PR. Please rebase against #77 and the additional commits will be removed.

File: src/common/string.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+#include <string>
+
+std::string operator ""_s(const char * str, size_t len) {
+    return std::string(str, len);
+}
```
Comment (psychocrypt) [inline src/common/string.cpp]: uhh I missed a EOF

File: src/common/string.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+#include <string>
+
+/** convert a C string into a C++ std::string
+ *
+ * @code{cpp}
+ * auto foo = "Hello World!"_s; // type of foo is std::string
+ * @endcode
+ */
+std::string operator ""_s(const char * str, size_t len);
```
Comment (psychocrypt) [inline src/common/string.hpp]: again no EOF, shame shame shame

**Comment by fireice-uk** [comment]
@psychocrypt You managed to change the commit hashes, so I had to cherry pick.  Please verify that I picked the right commits

**Comment by psychocrypt** [comment]
Uhh sry.

File: src/gen_multisig/gen_multisig.cpp [hunk 3]
```diff
@@ -208,7 +208,8 @@ int main(int argc, char *argv[])
 		genms::tr("This program generates a set of multisig wallets - use this simpler scheme only if all the participants trust each other"),
 		desc_params,
 		boost::program_options::positional_options_description(),
-		[](const std::string &s, bool emphasis) { tools::scoped_message_writer(emphasis ? epee::console_color_white : epee::console_color_default, true) << s; },
+		/*[](const std::string &s, bool emphasis) { tools::scoped_message_writer(emphasis ? epee::console_color_white : epee::console_color_default, true) << s; },
```
Comment (psychocrypt) [inline src/gen_multisig/gen_multisig.cpp]: If this is not needed anymore it should be removed else it is dead code.

**Comment by yoosofan** [comment]
Hi, hope you are fine. Why are you mentioning me on this ?

-- 
 ahmad yoosofan
 yoosofan@myfastmail.com

**Pull request closed without merging**